File: src/main/java/com/example/service/EmployeeService.java
package com.example.service;

import com.example.entity.Employee;
import com.example.exception.EmployeeNotFoundException; // Import your exception
import com.example.repository.EmployeeRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
 

import java.util.List;

@Service
public class EmployeeService {

    @Autowired
    private EmployeeRepository employeeRepository;

    // Add a new employee
    public Employee saveEmployee(Employee employee) {
        return employeeRepository.save(employee);
    }

    // Get all employees
    public List<Employee> getAllEmployees() {
        return employeeRepository.findAll();
    }

    // Get employee by ID
    public Employee getEmployeeById(Long id) {
        return employeeRepository.findById(id)
            .orElseThrow(() -> new EmployeeNotFoundException("Employee not found with id: " + id));
    }

    // Update employee
    public Employee updateEmployee(Long id, Employee updatedEmployee) {
        if (employeeRepository.existsById(id)) {
            updatedEmployee.setId(id);
            return employeeRepository.save(updatedEmployee);
        }
        return null;
    }

    // Delete employee
    public boolean deleteEmployee(Long id) {
        if (employeeRepository.existsById(id)) {
            employeeRepository.deleteById(id);
            return true;
        }
        return false;
    }
}

File: frontend/src/services/employeeService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/employees';

// Fetch all employees
export const getAllEmployees = () => axios.get(API_URL);

// Add a new employee
export const addEmployee = (employee) => axios.post(API_URL, employee);

// Update an existing employee
export const updateEmployee = (id, employee) => axios.put(`${API_URL}/${id}`, employee);

// Delete an employee
export const deleteEmployee = (id) => axios.delete(`${API_URL}/${id}`);
